qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -0,0 +1,140 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sparse Label Operator base class."""
+
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Mapping, Iterator
+from numbers import Number
+import cmath
+import numpy as np
+
+from qiskit.quantum_info.operators.mixins import LinearMixin, AdjointMixin, TolerancesMixin
+
+
+class SparseLabelOp(LinearMixin, AdjointMixin, TolerancesMixin, ABC):
+    """The Sparse Label Operator base class."""
+
+    def __init__(self, data: Mapping[str, complex], register_length: int, **kwargs):
+        """
+        Args:
+            data: Data for operator comprising string key to coeff value mapping
+            register_length: Length of register needed for data
+        """
+        self._data: Mapping[str, complex] = {}
+        if kwargs.get("_dont_copy", False):
+            self._data = data  # Store by reference
+        else:
+            self._data = dict(data.items())  # Store by value
+        self._register_length = register_length
```

Comment:
A `@property` for `register_length` should be provided, which will remove the need to access this private variable